public/js/NeighbourImages.js

```diff
@@ -6,14 +6,26 @@ var _ = require('lodash');
  * @constructor
  */
 function NeighbourImages() {
-    var self = this;
-
     this.neighbours = [];
     this.selectedImage = "";
     this.neighbourImages = [];
 
-    var divWidth = Math.round($('#image-column').width()) - 20;
-    var imgWidth = Math.round(divWidth/3) - 20;
+    // cache jquery selectors used frequency by gallery methods
+    // call these in the constructor so the DOM is traversed only once
+    // before caching.
```

Comment:
BOOM. Please copy-paste this almost verbatim into the code! =) (You don't have to write you were being stupid. =P)